### PR TITLE
Cookie and basic auth support

### PR DESCRIPTION
**Pull request by MuH3gPaB**
Added cookie and basic auth suppot.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -89,3 +89,19 @@ jobs:
...
+
+  check-single-with-basic-auth:
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+    runs-on: ${{ matrix.os }}
+    name: Check single, with cookie
```
Comment (Jtalk) [inline .github/workflows/check.yml]: The cookie option doesn’t seem to be actually set below. Can I ask you provide a test for the cookie too?

**Review by Jtalk — APPROVED** [review]
Thank you very much!

File: .github/workflows/check.yml (hunk 1 above)
Comment (MuH3gPaB) [reply .github/workflows/check.yml]: It's just a typo. Fixed.

I've tried to find any public test endpoint which requires cookie for auth, but unfortunately unsuccessful.
So I just checked it locally.

If you have any suitable endpoint or probably know other ways to test it, please, let me know.

Thank you!

**Review by Jtalk — APPROVED** [review]

File: .github/workflows/check.yml (hunk 1 above)
Comment (Jtalk) [reply .github/workflows/check.yml]: Thanks for the context, I see what the problem is now. 

I can't think of an easy way to test them either. Using the Cookie Get echo & verifying the output comes to mind, but the step doesn't provide any `output` as of yet. I'll see if I can fix it later.

Meanwhile, I'm happy to merge the PR as it is, thank you for your contribution!

**Pull request merged**